components/layout_2020/flow/mod.rs

```diff
@@ -1609,7 +1605,7 @@ impl PlacementState {
                 }
 
                 if self.next_in_flow_margin_collapses_with_parent_start_margin {
-                    debug_assert_eq!(self.current_margin.solve(), Length::zero());
+                    debug_assert_eq!(self.current_margin.solve().is_zero(), true);
```

Comment:
If the assert fails, I think `debug_assert_eq!(self.current_margin.solve(), Au::zero());` will provide the non-zero value, so it could be more useful.

Or otherwise, simplify to `debug_assert!(self.current_margin.solve().is_zero());`